baselines/FedPM/FedPM/main.py

```diff
@@ -0,0 +1,121 @@
+"""Create and connect the building blocks for your experiments; start the simulation.
+
+It includes processioning the dataset, instantiate strategy, specify how the global
+model is going to be evaluated, etc. At the end, this script saves the results.
+"""
+# these are the basic packages you'll need here
+# feel free to remove some if aren't needed
+
+import hydra
+import torch
+import sys
+import os
+import flwr
+from omegaconf import DictConfig, OmegaConf
+from client import FedPMClient
+
+from dataset import get_data_loaders
+from strategy import FedPMStrategy
+
+
+get_client = {
+    'fedpm': FedPMClient
+}
+
+
+@hydra.main(config_path="conf", config_name="base", version_base=None)
+def main(cfg: DictConfig) -> None:
+    """Run the baseline.
+
+    Parameters
+    ----------
+    cfg : DictConfig
+        An omegaconf object that stores the hydra config.
+    """
+    # 1. Print parsed config
+    print(OmegaConf.to_yaml(cfg))
+
+    # 2. Prepare your dataset
+    # here you should call a function in datasets.py that returns whatever is needed to:
+    # (1) ensure the server can access the dataset used to evaluate your model after
+    # aggregation
+    # (2) tell each client what dataset partitions they should use (e.g. a this could
+    # be a location in the file system, a list of dataloader, a list of ids to extract
+    # from a dataset, it's up to you)
+
+    trainloaders, valloaders, testloader = get_data_loaders(
+        dataset=cfg.get('dataset').get('name'),
+        nclients=NUM_CLIENTS,
+        batch_size=cfg.get('dataset').get('minibatch_size'),
+        classes_pc=cfg.get('dataset').get('classes_pc'),
+        split=cfg.get('dataset').get('split')
+    )
+
+    # 3. Define your clients
+    # Define a function that returns another function that will be used during
+    # simulation to instantiate each individual client
+
+    def client_fn(cid) -> flwr.client.Client:
+        trainloader = trainloaders[int(cid)]
+        valloader = valloaders[int(cid)]
+
+        return get_client[cfg.get('simulation').get('strategy')](
+            params=cfg,
+            client_id=cid,
+            train_data_loader=trainloader,
+            test_data_loader=valloader,
+            device=DEVICE,
+        )
+
+    # 4. Define your strategy
+    # pass all relevant argument (including the global dataset used after aggregation,
+    # if needed by your method.)
+    strategy = get_strategy[cfg.get('simulation').get('strategy')](
```

Comment:
Maybe here you want to make use of Hydra's `instantiate` functionality. See the hydra docs or the `FedProx` baseline i point you to below in my comment.